modules/resonator.h

```diff
@@ -0,0 +1,184 @@
+#pragma once
+#ifndef DSY_RESONATOR_H
+#define DSY_RESONATOR_H
+
+#include <stdint.h>
+#include <stddef.h>
+#include "dsp.h"
+#ifdef __cplusplus
+
+#define PI_POW_3 PI_F* PI_F* PI_F
+#define PI_POW_5 PI_POW_3* PI_F* PI_F
+
+#define kMaxNumModes 24
+#define kModeBatchSize 4
```

Comment:
make these private `static constexpr` in the relevant classes so that they're not accidentally overwritten or used outside of this module.